src/script/lua_api/l_util.cpp

```diff
@@ -131,9 +138,15 @@ int ModApiUtil::l_parse_json(lua_State *L)
 		}
 	}
 
+
 	if (!push_json_value(L, root, nullindex)) {
-		errorstream << "Failed to parse json data, "
-			<< "depth exceeds lua stack limit" << std::endl;
+		const char *err_stack_depth = "depth exceeds lua stack limit";
+		if (return_error) {
+			lua_pushnil(L);
+			lua_pushstring(L, err_stack_depth);
+			return 2;
+		}
+		errorstream << "Failed to parse json data, " << err_stack_depth << std::endl;
 		errorstream << "data: \"" << jsonstr << "\"" << std::endl;
```

Comment:
this should be made consistent with the other one.
tho i don't see the point in this errorstream/warningstream distinction, so you can get rid of that.

Reply:
sure. done.